### PR TITLE
Gap filling

### PR DESCRIPTION
**Pull request by sbiquard**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,24 +1,14 @@
...
 
 [project]
 name = 'furax'
-authors = [
-    {name = 'Pierre Chanial', email = 'chanial@apc.in2p3.fr'},
-]
-maintainers = [
-    {name = 'Pierre Chanial', email = 'chanial@apc.in2p3.fr'},
-]
+authors = [{ name = 'Pierre Chanial', email = 'chanial@apc.in2p3.fr' }]
```
Comment (pchanial) [inline pyproject.toml:7]: Il faut t'ajouter dans les auteurs.

**Review by pchanial — CHANGES_REQUESTED** [review]

File: src/furax/detectors.py [hunk 2]
```diff
@@ -22,5 +26,29 @@ def __init__(
         coords /= length
         self.coords = jax.device_put(coords)
 
+        # generate fake names for the detectors
+        # TODO(simon): accept user-defined names
+        det_indices = (_.astype(np.str_) for _ in np.indices(self.shape, sparse=True))
+        names = reduce(lambda a, b: a + '_' + b, det_indices, 'FAKE_DET')
```
[the remark this answers is not in the document]
Comment (sbiquard) [reply src/furax/detectors.py]: Great suggestion, done in https://github.com/CMBSciPol/furax/pull/4/commits/90edb4340546e8adbf819b18a5634f89b2e6953f

File: src/furax/detectors.py [hunk 3]
```diff
@@ -22,5 +26,29 @@ def __init__(
...
         return int(np.prod(self.shape))
+
+    def _ids(self) -> UInt32[Array, '...']:
+        # vectorized hashing + converting to int + keeping only 7 bytes
+        name_to_int = np.vectorize(lambda s: int(sha1(s.encode()).hexdigest(), 16) & 0xEFFFFFFF)
+        # return detectors IDs as unsigned 32-bit integers
+        ids: UInt32[Array, '...'] = jnp.uint32(name_to_int(self.names))
+        return ids
+
+    def split_key(self, key: PRNGKeyArray) -> Shaped[PRNGKeyArray, ' _']:
```
[the remark this answers is not in the document]
Comment (sbiquard) [reply src/furax/detectors.py]: Done in https://github.com/CMBSciPol/furax/pull/4/commits/90edb4340546e8adbf819b18a5634f89b2e6953f

File: src/furax/detectors.py [hunk 4]
```diff
@@ -22,5 +26,29 @@ def __init__(
...
+        """Folds the detector names in a random key to generate a key array."""
```
[the remark this answers is not in the document]
Comment (sbiquard) [reply src/furax/detectors.py]: Also done in https://github.com/CMBSciPol/furax/pull/4/commits/90edb4340546e8adbf819b18a5634f89b2e6953f

File: src/furax/preprocessing/gap_filling.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+default_fft_size = SymmetricBandToeplitzOperator._get_default_fft_size
+
+__all__ = [
+    'GapFillingOperator',
+]
+
+
+class GapFillingOperator(equinox.Module):
+    """Class for filling masked time samples with a constrained noise realization.
```
[the remark this answers is not in the document]
Comment (sbiquard) [reply src/furax/preprocessing/gap_filling.py:20]: See https://github.com/CMBSciPol/furax/pull/4/commits/668be101160e0c8bf71cf1deb7c49ba9b1b5ab1a

**Pull request merged**